query/src/main/java/io/keen/client/java/Query.java

```diff
@@ -0,0 +1,416 @@
+package io.keen.client.java;
+
+import java.util.Map;
+import java.util.List;
+import java.util.HashMap;
+import java.util.ArrayList;
+
+/**
+ * Created by claireyoung on 5/18/15.
+ */
+public class Query {
+
+    private QueryType queryType;
+
+    // required for all queries
+    private String eventCollection;
+
+    // required by most queries
+    private String targetProperty;
+
+    // optional
+    private List<Map<String, Object>> filters;
+    private String interval;    // requires timeframe to be set
+    private String timezone;
+    private ArrayList<String> groupBy;
+    private Integer maxAge; // integer greater than 30 seconds: https://keen.io/docs/data-analysis/caching/
+
+    // required by the Percentile query
+    private Double percentile;  // 0-100 with two decimal places of precision for example, 99.99
+
+
+    /**
+     * Constructs the map to pass to the JSON handler, so that the proper required
+     * and optional Query arguments can be sent out to the server.
+     *
+     * @return The JSON object map.
+     */
+    public Map<String, Object> ConstructQueryArgs() {
```

Comment:
I see at least one other instance of an initial capital (down below in AreParamsValid). You may want to do a quick scan yourself for more of these, just in case.